src/elements/content-sidebar/SidebarPanels.js

```diff
@@ -96,31 +103,45 @@ const SidebarPanels = ({
                 enabled={hasActivityFeed}
                 path={`/${SIDEBAR_VIEW_ACTIVITY}`}
                 pathFallback={`/${SIDEBAR_VIEW_DETAILS}`}
-                render={() =>
+                render={({ match }) =>
                     isOpen && (
-                        <LoadableActivitySidebar
-                            currentUser={currentUser}
-                            file={file}
-                            onVersionHistoryClick={onVersionHistoryClick}
-                            {...activitySidebarProps}
-                            startMarkName={MARK_NAME_JS_LOADING_ACTIVITY}
-                        />
+                        <Switch>
+                            <Route exact path={match.path}>
+                                <LoadableActivitySidebar
+                                    currentUser={currentUser}
+                                    file={file}
+                                    onVersionHistoryClick={onVersionHistoryClick}
+                                    startMarkName={MARK_NAME_JS_LOADING_ACTIVITY}
+                                    {...activitySidebarProps}
+                                />
+                            </Route>
+                            <Route exact path={`${match.path}/versions`}>
+                                <LoadableVersionsSidebar fileId={fileId} key={fileId} />
```

Comment:
What benefit does providing the `key` here have?

Reply:
It forces a remount of the component so we can avoid tracking whether the fileId changed.